### PR TITLE
Fire and forget api needs no reply

### PR DESCRIPTION
**Pull request by sandyatharman**
Max number of replies that can be pending for each connection count is limited to 127 and 50000 in case of SYSTEM bus and SESSION bus respectively. Currently when an api call of type fireAndForget is made, it's DBUS_HEADER_FLAG_NO_REPLY_EXPECTED flag is false so the count for pending call is incremented for every call of type fireAndForget and since no reply is sent for such calls this count will keep on increasing till it reaches the configuration specified limit. Once the limit is reached, no more calls can be made. Since fireAndForget api calls don't expect any reply/response we can set the flag DBUS_HEADER_FLAG_NO_REPLY_EXPECTED to FALSE.

### CODE REVIEW

**Comment by sandyatharman** [comment]
Changes are only in the commit id ea2846be72c3609316f612c231b218b0ca3e3185

File: src/CommonAPI/DBus/DBusMessage.cpp [hunk 1]
```diff
@@ -292,5 +292,9 @@ bool DBusMessage::isSignalType() const {
     return (getType() == Type::Signal);
 }
 
+void DBusMessage::setNoReplyExpected(bool replyNotExpected) {
+    dbus_message_set_no_reply (message_, (0 != replyNotExpected) ? TRUE : FALSE);
```
Comment (gunnarx) [inline src/CommonAPI/DBus/DBusMessage.cpp]: This is very convoluted.  
Set "No reply", using the variable named "Not Expected" by checking if it is  "Not equal to zero (false)" (i.e true)...

If you're sure that dbus_message_set_no_reply can use the bool type value directly then maybe just put it in directly?   If not, calling the values TRUE and FALSE do not make it any safer.  I would then pass in an integer value if you are unsure.

On the other hand, I would rename all of it to setReplyExpected and let the parameter be true if reply expected.

File: src/CommonAPI/DBus/DBusMessage.cpp [hunk 2]
```diff
@@ -292,5 +292,9 @@ bool DBusMessage::isSignalType() const {
     return (getType() == Type::Signal);
 }
 
+void DBusMessage::setNoReplyExpected(bool replyNotExpected) {
```
Comment (gunnarx) [inline src/CommonAPI/DBus/DBusMessage.cpp]: It's so short that an inline function might be preferable?

File: include/CommonAPI/DBus/DBusMessage.hpp [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 #include <dbus/dbus.h>
 
-#ifdef WIN32
+#ifdef _WIN32
```
Comment (gunnarx) [inline include/CommonAPI/DBus/DBusMessage.hpp]: This looks unrelated.  Is it necessary?

File: include/CommonAPI/DBus/DBusMessage.hpp [hunk 4]
```diff
@@ -94,6 +94,7 @@ class COMMONAPI_EXPORT DBusMessage {
     bool setDestination(const char* destination);
 
     void setSerial(const unsigned int serial) const;
+    void setNoReplyExpected(bool replyNotExpected);
```
Comment (gunnarx) [inline include/CommonAPI/DBus/DBusMessage.hpp]: How about setReplyExpected instead? (See later comment).

Unless you are sure it will never be called to set reply expected, always to set reply not expected.  In that case the parameter can be dropped.

**Comment by gunnarx** [comment]
This looks like a great and important change @sandyatharman.  Thanks from me (but I am not component maintainer) for providing it.

I made some review comments for the actual code in commit https://github.com/GENIVI/capicxx-dbus-runtime/pull/5/commits/ea2846be72c3609316f612c231b218b0ca3e3185

General comments - the summary line is too long for a git commit:

"Setting the DBUS_HEADER_FLAG_NO_REPLY_EXPECTED flag to TRUE for each dbus message of type fireAndForget"
instead:
"Set no reply expected for fire-and-forget messages"

Then there can be more text below.

For this component the maintainer decides the policy, but here is what we [require for GDP](https://at.projects.genivi.org/wiki/display/PROJ/Rules+for+Commit+Messages) which might be of some help.

I don't understand the commit named "Add files via upload": https://github.com/GENIVI/capicxx-dbus-runtime/pull/5/commits/52020a0e758cd6bba6c939a1038cb876f7435dd8 because it seems to add whole new file - why?

**Comment by GenivivSOMEIPmaintainer** [comment]
Has been integrated in CommonAPI 3.1.12.
